device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -151,7 +151,18 @@ else if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.X509
                     clientId += "/" + moduleId;
                 }
 
-                this.iotHubUserName = this.config.getIotHubHostname() + "/" + clientId + "/" + API_VERSION + "&" + clientUserAgentIdentifier;
+                String apiVersion;
+                String modelId = this.config.getModelId();
+                if(modelId == null)
```

Comment:
`modelId == null || modelId.isEmpty()`

Reply:
Looks like java does not have a single function for this :)